challenge3.js

```diff
@@ -0,0 +1,47 @@
+// import readline from 'readline';
+const readline = require('readline');
+
+const rl = readline.createInterface({
+    input: process.stdin,
+    output: process.stdout
+});
+
+function getBirthday() {
+    return new Promise(resolve => {
+      rl.question('Please enter your birthday in the format of MM/DD/YYYY: ', answer => {
+        resolve(answer);
+      });
+    });
+}
+
+function validateBirthday(birthday) {
+    const birthdayDate = new Date(birthday);
+    const currentDate = new Date();
+    if (birthdayDate > currentDate) {
+      console.log('Please enter a valid birthday in the past.');
+      return false;
+    }
+    return true;
+}
+
+async function calculateAge() {
```

Comment:
Could you please explain why you have used `async` here?

Reply:
Yes ,sure
The reason why async is used  to handle asynchronos operations, like the user inputs their birthday. The getBirthday() function uses rl.question  which takes input from the user asynchronously (program cannot continue executing until the user input is received ) .

when we use async we should use await to wait for the completion of the asynchronous operation .